client/src/pages/Join.js

```diff
@@ -62,20 +63,35 @@ function Join() {
     const socketRef = useRef();
     socketRef.current = io.connect("/");
     const [matchId, setMatchId] = useState("");
+    const [snackbarOpen, setSnackbarOpen] = useState(false);
+    const [joinError, setJoinError] = useState("");
 
     const onTextChange = (e) => {
         setMatchId(e.target.value);
     };
 
-    const submitMatchId = (e) => {
+    const submitMatchId = async (e) => {
         e.preventDefault();
-        // TODO: call api route for joining match
-        socketRef.current.emit("join-match", { matchId });
-        socketRef.current.on("join-game-engine", (game) => {
-            setMatchState(game);
-            socketRef.current.disconnect();
-        });
-        return history.push(`/join/${matchId}`);
+        try {
+          const response = await fetch(`/api/match/join/${matchId}`, {
+            method: 'POST',
+            headers: { 'Content-Type': 'application/json' },
+          });
+          const data = await response.json();
+          if (response.status === '200') {
+            socketRef.current.emit('join-match', { matchId });
+            socketRef.current.on('join-game-engine', (game) => {
+              setMatchState(game);
+              socketRef.current.disconnect();
```

Comment:
Instead of disconnect, you want to disconnect after the user leaves the application to limit reconnecting every component